### PR TITLE
feature 22/nwp_scripts

### PR DESCRIPTION
**Pull request by fossell**
## Expected Differences ##

- [x] Do these changes modify the system output in any way? **[No]**</br>
These changes add new output.

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Tested on Cheyenne using Mike's wrfcloud conda environment available on disk. run.py was run and output in test/ungrib and test/metgrid confirmed looked reasonable using ncview.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Reviewers can test on Cheyenne and should place files and source code in appropriate places, e.g. link WPSV4 and WRFV4 to wrfcloud/python/src/wrfcloud/runtime/ and place geo_em.d01.nc file in wrfcloud/python/src/wrfcloud/runtime/domains/test.
On Cheyenne:
`ln -sf /glade/p/ral/jntp/CCproject/wrfcloud/WRFV4.4_intel_dmpar/ WRFV4`
`ln -sf /glade/p/ral/jntp/CCproject/wrfcloud/WPSV4.4_intel_serial_large-file/ WPSV4`
`cp /glade/p/ral/jntp/CCproject/wrfcloud/wpsprd/geo_em.d01.nc $PATH_TO_CLONED_REPO/wrfcloud/python/src/wrfcloud/runtime/domains/test`

To test on an AWS cluster, need to address topdir and relative paths, e.g. WPS and WRF are placed in /home/ec2-user, and output should go to /data. Additional work is need to generalize this for use on the cluster.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No]**
No documentation was added for these mods. This shouldn't be something the user interacts with.

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **[Friday, August 5, 2022]**.</br>

## Pull Request Checklist ##
- [x] Review the source issue metadata (labels, project, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Project**
Select: **Milestone** as the version that will include these changes
Select: **Development** to link to the original development issue.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: python/src/wrfcloud/runtime/metgrid.py [hunk 1]
```diff
@@ -5,13 +5,46 @@
...
+    """Gets all input files necessary for running ungrib"""
+
+    logger.debug('Getting geo_em file(s)')
+    # Get the number of domains from namelist
+    # Assumes geo_em files are in local path/domains/expn_name. TODO: Make pull from S3
+    for domain in range(1, namelist['share']['max_dom'] + 1):
+        os.symlink(f'{runinfo.staticdir}/geo_em.d{domain:02d}.nc', f'geo_em.d{domain:02d}.nc')
+
+    logger.debug('Linking metgrid dir for tables')
+    os.symlink(f'{runinfo.wpsdir}/metgrid', f'metgrid')
```
Comment (hahnd) [inline python/src/wrfcloud/runtime/metgrid.py:28]: Second parameter to symlink does not have any variable substitution and does not need to be an f-string.  (f'metgrid' can be changed to 'metgrid').

File: python/src/wrfcloud/runtime/ungrib.py [hunk 2]
```diff
@@ -61,12 +61,25 @@ def get_files(runinfo: RunInfo, logger: Logger, namelist: Namelist) -> None:
...
+    logger.debug('Getting VTable.GFS')
+    os.symlink(f'{runinfo.wpsdir}/ungrib/Variable_Tables/Vtable.GFS', f'Vtable')
 
 
+    logger.debug('Linking ungrib.exe to ungrib working directory')
+    os.symlink(f'{runinfo.wpsdir}/ungrib/ungrib.exe', f'ungrib.exe')
+
+def run_ungrib(runinfo: RunInfo, logger: Logger, namelist: Namelist) -> None:
+    """Executes the ungrib.exe program"""
+    logger.debug('Executing ungrib.exe')
```
[the remark this answers is not in the document]
Comment (fossell) [reply python/src/wrfcloud/runtime/ungrib.py:77]: I had this in get_files for some reason. I moved it to the run function like metgrid.

File: python/src/wrfcloud/runtime/metgrid.py [hunk 3]
```diff
@@ -5,13 +5,46 @@
...
+
+def run_metgrid(runinfo: RunInfo, logger: Logger, namelist: Namelist) -> None:
+    """Executes the metgrid.exe program"""
+
+    logger.debug('Linking metgrid.exe to metgrid working directory')
+    os.symlink(f'{runinfo.wpsdir}/metgrid/metgrid.exe', f'metgrid.exe')
+
+    logger.debug('Executing metgrid.exe')
+    ungrib_cmd ='./metgrid.exe >& metgrid.log'
+    os.system(ungrib_cmd)
```
[the remark this answers is not in the document]
Comment (fossell) [reply python/src/wrfcloud/runtime/metgrid.py:45]: I need to discuss with others about best ways to address this. To my knowledge no standard error codes but perhaps there are and I'm not aware. Typically I've checked for success by grepping for Success in the log files, checking for the proper output files and that they're non zero size, etc. I will make this an issue to tackle with the group.

**Review by michelleharrold — APPROVED** [review]
I was able to successfully run the nwp_scripts branch on Cheyenne for the canned data 'test' case -- producing the expected ungrib and metgrid output! 

As a note: The first attempt did fail (user error with loading necessary modules), but we have adding error checking on the to-do list to address this (Issue #26).

**Pull request merged**